stores/hostdb.go

```diff
@@ -426,6 +499,155 @@ func (ss *SQLStore) Host(ctx context.Context, hostKey types.PublicKey) (hostdb.H
 	}, nil
 }
 
+func (ss *SQLStore) HostInfo(ctx context.Context, autopilotID string, hk types.PublicKey) (hi api.HostInfo, err error) {
+	err = ss.db.Transaction(func(tx *gorm.DB) error {
+		var entity dbHostInfo
+		if err := tx.
+			Model(&dbHostInfo{}).
+			Where("db_autopilot_id = (?)", gorm.Expr("SELECT id FROM autopilots WHERE identifier = ?", autopilotID)).
+			Where("db_host_id = (?)", gorm.Expr("SELECT id FROM hosts WHERE public_key = ?", publicKey(hk))).
+			Preload("DBHost").
+			First(&entity).
+			Error; errors.Is(err, gorm.ErrRecordNotFound) {
+			if err := tx.
```

Comment:
I think it's meaningful information to return either `ErrAutopilotNotFound` or `ErrHostNotFound` instead of returning `ErrHostInfoNotFound` in all cases.